src/XMTP.types.ts

```diff
@@ -44,6 +44,34 @@ export type RemoteAttachmentContent = RemoteAttachmentMetadata & {
   url: string;
 };
 
+// This contains a message that has been prepared for sending.
+// It contains the message ID and the URI of a local file
+// containing the payload that needs to be published.
+// See Conversation.sendPreparedMessage() and Client.sendPreparedMessage()
+//
+// For native integrations (e.g. if you have native code for a robust
+// pending-message queue in a background task) you can load the referenced
+// `preparedFileUri` as a serialized `PreparedMessage` with the native SDKs.
+// The contained `envelopes` can then be directly `.publish()`ed with the native `Client`.
+//   e.g. on iOS:
+//    let preparedFileUrl = URL(string: preparedFileUri)
+//    let preparedData = try Data(contentsOf: preparedFileUrl)
+//    let prepared = try PreparedMessage.fromSerializedData(preparedData)
+//    try await client.publish(envelopes: prepared.envelopes)
+//   e.g. on Android:
+//     val preparedFileUri = Uri.parse(preparedFileUri)
+//     val preparedData = contentResolver.openInputStream(preparedFileUrl)!!
+//         .use { it.buffered().readBytes() }
+//     val prepared = PreparedMessage.fromSerializedData(preparedData)
+//     client.publish(envelopes = prepared.envelopes)
+//
+// You can also stuff the `preparedData` elsewhere (e.g. in a database) if that
+// is more convenient for your use case.
```

Comment:
Some apps implement a native background task for robust pending-message queues (e.g. to gracefully handle outages and restarts outside of a JS context). Here are some pieces of a recipe to handle publishing these prepared messages using the native SDKs.